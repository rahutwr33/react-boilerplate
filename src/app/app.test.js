// app.test.js
import React from 'react'
import { Router, MemoryRouter } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import '@testing-library/jest-dom/extend-expect'
import App from './app'
import { shallow, mount } from 'enzyme';

describe('Render App componet', () => {
    const history = createMemoryHistory()
    const wrapper = shallow(
        <Router history={history}>
            <App />
        </Router>
    )
    test('full app rendering', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('full app rendering/navigating', () => {
        const container = mount(
            <MemoryRouter initialEntries={['/signup']} >
                <App />
            </MemoryRouter>
        )
        expect(container.find('#signup'))
    })

    test('invalid path should redirect to 404', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/random']} >
                <App />
            </MemoryRouter>
        );

        expect(wrapper.find('#nomatch'));
    });
})
