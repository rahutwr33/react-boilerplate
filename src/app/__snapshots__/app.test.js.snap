// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App componet full app rendering 1`] = `ShallowWrapper {}`;
